packages/openapi3-parser/lib/parser/oas/parseMediaTypeObject.js

```diff
@@ -89,19 +89,28 @@ const createTextMessageBodyAsset = R.curry((namespace, mediaType, value) => {
   return asset;
 });
 
-function parseExample(namespace, mediaType) {
-  const createExampleNotJSONWarning = createWarning(namespace,
+function parseExample(context, mediaType) {
+  const { namespace } = context;
+
+  const createExampleNotSupportedForMediaTypeWarning = createWarning(namespace,
     `'${name}' 'example' is not supported for media type '${mediaType}'`);
 
+  const createExampleNotStringWarning = createWarning(namespace,
+    `'${name}' 'example' should be a string for media type '${mediaType}'`);
+
   const isTextBasedType = R.anyPass([
     () => isTextMediaType(mediaType),
     () => isXMLMediaType(mediaType),
   ]);
 
+  const parseTextExample = pipeParseResult(namespace,
+    R.unless(isString, createExampleNotStringWarning),
+    createTextMessageBodyAsset(namespace, mediaType));
+
   return R.compose(R.cond([
     [() => isJSONMediaType(mediaType), createJSONMessageBodyAsset(namespace, mediaType)],
-    [() => isTextBasedType(mediaType), createTextMessageBodyAsset(namespace, mediaType)],
-    [R.T, createExampleNotJSONWarning],
+    [() => isTextBasedType(mediaType), parseTextExample],
+    [R.T, createExampleNotSupportedForMediaTypeWarning],
```

Comment:
This old name doesn't reflect what the method does anymore, I think the name pre-dates adding support for text based media types.